repos/c2sm/packages/icon/package.py

```diff
@@ -611,19 +638,46 @@ def configure_args(self):
                     for d in self._get_cuda_ccbin_link_paths(gcc, 'libstdc++')
                 ]
                 cuda_host_compiler_stdcxx_libs.append('-lstdc++')
+                config_vars['NVCFLAGS'].extend(
+                    ['-ccbin {0}'.format(cuda_host_compiler)])
             else:
                 cuda_host_compiler = self.compiler.cxx
                 cuda_host_compiler_stdcxx_libs = self.compiler.stdcxx_libs
 
-            config_vars['NVCFLAGS'].extend([
-                '-ccbin {0}'.format(cuda_host_compiler), '-g', '-O3',
-                '-arch=sm_{0}'.format(gpu)
-            ])
+            config_vars['NVCFLAGS'].extend(
+                ['-g', '-O3', '-arch=sm_{0}'.format(gpu)])
             # cuda_host_compiler_stdcxx_libs might contain compiler-specific
             # flags (i.e. not the linker -l<library> flags), therefore we put
             # the value to the config_flags directly.
             config_vars['LIBS'].extend(cuda_host_compiler_stdcxx_libs)
 
+        # Check for DSL variants and set corresponding Liskov options
+        dsl = self.spec.variants['dsl'].value
+        if dsl != 'none':
+            tty.warn('values of dsl:::: {}'.format(dsl))
+            if 'substitute' in dsl:
+                config_args.append('--enable-liskov=substitute')
+            elif 'verify' in dsl:
+                config_args.append('--enable-liskov=verify')
+            elif 'serialize' in dsl:
+                raise error.UnsupportedPlatformError(
+                    'serialize mode is not supported yet by icon-liskov')
+
+            if 'lam' in dsl:
+                config_args.append('--enable-dsl-local')
+            if 'nvtx' in dsl:
+                config_args.append('--enable-nvtx')
+            if 'fused' in dsl:
+                raise error.UnsupportedPlatformError(
+                    'liskov does not support fusing just yet')
+
+            config_vars['LOC_GT4PY'].append(self.spec['py-gt4py'].prefix)
+            config_vars['LOC_ICON4PY'].append(
+                os.path.join(self.spec['py-icon4py'].prefix,
+                             'lib/python3.10/site-packages/icon4py'))
```

Comment:
Since we hardcode the pip-created path `lib/python3.10` I would put a conflict for `+dsl` to ensure we only use python3.10.

Something like
```python
conflict(when= +dsl ^python@:3.9,python@3:11)
```
Not sure about the exact syntax.

Reply:
good idea.